calico-vpp-agent/policy/policy_server.go

```diff
@@ -317,87 +369,126 @@ func (s *Server) Stop() {
 			} else {
-				s.log.Errorf("Got message %+v but not in syncing or synced state", m)
-				conn.Close()
-				s.felixRestarted <- true
-				return
+				s.log.Errorf("Got tunnel change %+v but not in syncing or synced state", tunnelChange)
+				continue
 			}
-			switch m := msg.(type) {
-			case *proto.IPSetUpdate:
-				err = s.handleIpsetUpdate(m, pending)
-			case *proto.IPSetDeltaUpdate:
-				err = s.handleIpsetDeltaUpdate(m, pending)
-			case *proto.IPSetRemove:
-				err = s.handleIpsetRemove(m, pending)
-			case *proto.ActivePolicyUpdate:
-				err = s.handleActivePolicyUpdate(m, pending)
-			case *proto.ActivePolicyRemove:
-				err = s.handleActivePolicyRemove(m, pending)
-			case *proto.ActiveProfileUpdate:
-				err = s.handleActiveProfileUpdate(m, pending)
-			case *proto.ActiveProfileRemove:
-				err = s.handleActiveProfileRemove(m, pending)
-			case *proto.HostEndpointUpdate:
-				err = s.handleHostEndpointUpdate(m, pending)
-			case *proto.HostEndpointRemove:
-				err = s.handleHostEndpointRemove(m, pending)
-			case *proto.WorkloadEndpointUpdate:
-				err = s.handleWorkloadEndpointUpdate(m, pending)
-			case *proto.WorkloadEndpointRemove:
-				err = s.handleWorkloadEndpointRemove(m, pending)
-			case *proto.HostMetadataUpdate:
-				err = s.handleHostMetadataUpdate(m, pending)
-			case *proto.HostMetadataRemove:
-				err = s.handleHostMetadataRemove(m, pending)
-			case *proto.IPAMPoolUpdate:
-				err = s.handleIpamPoolUpdate(m, pending)
-			case *proto.IPAMPoolRemove:
-				err = s.handleIpamPoolRemove(m, pending)
-			case *proto.ServiceAccountUpdate:
-				err = s.handleServiceAccountUpdate(m, pending)
-			case *proto.ServiceAccountRemove:
-				err = s.handleServiceAccountRemove(m, pending)
-			case *proto.NamespaceUpdate:
-				err = s.handleNamespaceUpdate(m, pending)
-			case *proto.NamespaceRemove:
-				err = s.handleNamespaceRemove(m, pending)
-			default:
-				s.log.Warnf("Unhandled message from felix: %v", m)
+			state := s.currentState(pending)
+			for _, h := range state.HostEndpoints {
+				if tunnelChange.ChangeType == connectivity.AddChange {
+					newTunnel := true
+					for _, v := range h.TunnelSwIfIndexes {
+						if v == tunnelChange.Swifindex {
+							newTunnel = false
+						}
+					}
+					if newTunnel {
+						h.TunnelSwIfIndexes = append(h.TunnelSwIfIndexes, tunnelChange.Swifindex)
+						s.log.Infof("Configuring policies on added tunnel [%d]", tunnelChange.Swifindex)
+						err = s.vpp.ConfigurePolicies(tunnelChange.Swifindex, h.currentForwardConf)
+						if err != nil {
+							s.log.Errorf("cannot configure policies on tunnel interface %d", tunnelChange.Swifindex)
+						}
+					}
+				} else { // delete case
+					s.log.Infof("Tunnel [%d] is deleted", tunnelChange.Swifindex)
+					for index, existingSwifindex := range h.TunnelSwIfIndexes {
+						if existingSwifindex == tunnelChange.Swifindex {
+							h.TunnelSwIfIndexes = append(h.TunnelSwIfIndexes[:index], h.TunnelSwIfIndexes[index+1:]...)
```

Comment:
Might be nice to add a comment saying that we don't delete the policies because they are auto-deleted when interfaces are removed, otherwise this could look like a bug.